cookbooks/continental_extension.prm

```diff
@@ -0,0 +1,256 @@
+subsection Compositional initial conditions
+  set Model name = function
+  subsection Function
+    set Variable names      = x,y
+    set Function expression = if(y>=80.e3, 1, 0); \
+                              if(y<80.e3 && y>=70.e3, 1, 0); \
+                              if(y<70.e3 && y>-100.e3,1, 0); \ 
+                              if(y<68.e3 && y>60.e3 && x>=198.e3 && x<=202.e3 , 1, 0);
+  end
+end
+
+# Temperature boundary conditions
+# Top and bottom (fixed) temperatures are consistent with the initial temperature field
+subsection Boundary temperature model
+  set Model name = box
+
+  subsection Box
+    set Bottom temperature = 1573
+    set Left temperature   =  273
+    set Right temperature  =  273
+    set Top temperature    =  273
+
+  end
+end
+
+# Initial temperature field
+# Typical continental geotherm based equations 4-6 from:
+#   D.S. Chapman (1986), "Thermal gradients in the continental crust",
+#   Geological Society of London Special Publications, v.24, p.63-70.
+# The initial constraints are:
+#   Layer Surface Temperature - upper crust (ts1) = 273 K; 
+#                               mantle      (ts3) = 823 K;  
+#   Model Base Temperature - (tb) = 1573 K;
+#   Heat Production - upper crust (A) = 1.5e-6 W/m^3; 
+#   Thermal Conductivity - upper crust (k1) = 2.5 (W/(m K)); 
+#                          lower crust (k2) = 2.5 (W/(m K)); 
+#                          mantle      (k3) = 3.3 (W/(m K));
+# To satisfy these constraints, the following values are required:
+#   Layer Surface Heat Flow - upper crust (qs1) = 0.065357 W/m^2; 
+#                             lower crust (qs2) = 0.035357 W/m^2; 
+#                             mantle      (qs3) = 0.035357 W/m^2;
+#   Temperature - base of upper crust (ts2) = 681.5714
+subsection Initial conditions
+  set Model name = function
+  subsection Function
+    set Variable names = x,y
+    set Function constants = h=100e3,ts1=273,ts2=681.5714,ts3=823., \
+                                     k1=2.5,k2=2.5,k3=3.3,A=1.5e-6, \
+                             qs1=0.0653571,qs2=0.035357,qs3=0.035357,qb3=0.035357
+    set Function expression = if( (h-y)<=20.e3, \
+                                  ts1 + (qs1/k1)*(h-y) - (A*(h-y)*(h-y))/(2.0*k1), \
+                                  if((h-y)>20.e3 && (h-y)<=30.e3, ts2 + (qs2/k2)*(h-y-20.e3),\
+                                  ts3 + (qs3/k3)*(h-y-30.e3)));
+  end
+end
+
+# Internal heating (only internal heating for crust)
+# Assuming a very large decay time, the heat production (W/m^3) is effectively:
+#   radioactive_heating_rate (W/kg) * density (kg/m^3)
+# For a reference crustal density of 2800 kg/m^3 and 1.5e-6 W/m^3 heating production,
+# the heating rate (assuming a single element) is 1.5e-6/2800. = 5.357142857e-10.
+subsection Heating model
+  set Model name = radioactive decay
+  subsection Radioactive decay
+    set Number of elements    = 1
+    set Heating rates                 = 5.357142857e-10
```

Comment:
This is a very specific number that I doubt you can measure to this accuracy. I think it would be sufficient to specify it here and in the comment above to 3 digits. After all, it's computed from numbers that you actually only know to two digits.